src/compiler/sys.ts

```diff
@@ -380,8 +369,35 @@ namespace ts {
                 }
             }
 
-            function getCanonicalPath(path: string): string {
-                return useCaseSensitiveFileNames ? path : path.toLowerCase();
+            function getAccessibleFileSystemEntries(path: string): FileSystemEntries {
+                try {
+                    const entries = _fs.readdirSync(path || ".").sort();
+                    const files: string[] = [];
+                    const directories: string[] = [];
+                    for (const entry of entries) {
+                        // This is necessary because on some file system node fails to exclude
+                        // "." and "..". See https://github.com/nodejs/node/issues/4002
+                        if (entry === "." || entry === "..") {
+                            continue;
+                        }
+                        const name = combinePaths(path, entry);
+                        const stat = _fs.statSync(name);

```

Comment:
May want to put a `try..catch` around the call to `statSync`. If we don't, then we'll fail to return the contents of a directory if even one entry in the directory has more restrictive permissions that would cause `statSync` to fail.